src/input/mod.rs

```diff
@@ -850,42 +850,97 @@ impl State {
+                                                                                .get_pointer()
+                                                                                .unwrap()
+                                                                                .set_grab(
+                                                                                    state, target, serial,
+                                                                                    focus,
+                                                                                );
+                                                                        }
+                                                                    }
+                                                                },
+                                                                smithay::backend::input::MouseButton::Right => {
+                                                                    let res = (|| {
+                                                                        let target_elem = shell.element_for_surface(&surface)?;
+                                                                        let geom = shell
+                                                                            .space_for(target_elem)?
+                                                                            .element_geometry(target_elem)?
+                                                                            .to_f64();
+                                                                        let center = geom.loc + geom.size.downscale(2.0);
+                                                                        let offset = center.to_global(&output) - pos;
+                                                                        let edge = match (offset.x > 0.0, offset.y > 0.0) {
+                                                                            (true, true) => ResizeEdge::TOP_LEFT,
+                                                                            (false, true) => ResizeEdge::TOP_RIGHT,
+                                                                            (true, false) => ResizeEdge::BOTTOM_LEFT,
+                                                                            (false, false) => ResizeEdge::BOTTOM_RIGHT
+                                                                        };
+                                                                        shell.resize_request(
+                                                                            &surface,
+                                                                            &seat_clone,
+                                                                            serial,
+                                                                            edge
+                                                                        )
+                                                                    })();
```

Comment:
While I certainly see the appeal of the `?` operator, I don't think we need this rather ugly closure here. `let`-`else` and `and_then` work perfectly fine here, so I would rather prefer something like this:

```suggestion
                                                                    let Some(target_elem) = shell.element_for_surface(&surface) else { return };
                                                                    let Some(geom) = shell
                                                                            .space_for(target_elem).and_then(|space| space.element_geometry(target_elem).to_f64()) else { return };
                                                                    let center = geom.loc + geom.size.downscale(2.0);
                                                                    let offset = center.to_global(&output) - pos;
                                                                    let edge = match (offset.x > 0.0, offset.y > 0.0) {
                                                                        (true, true) => ResizeEdge::TOP_LEFT,
                                                                        (false, true) => ResizeEdge::TOP_RIGHT,
                                                                        (true, false) => ResizeEdge::BOTTOM_LEFT,
                                                                        (false, false) => ResizeEdge::BOTTOM_RIGHT
                                                                    };
                                                                    let res = shell.resize_request(
                                                                        &surface,
                                                                        &seat_clone,
                                                                        serial,
                                                                        edge
                                                                    );
```